relay-server/src/utils/rate_limits.rs

```diff
@@ -766,6 +990,24 @@ where
             }
         }
 
+        // Handle span attachments (part 2)
+        // If we don't yet have limits for span attachments (derived from the span limits)
+        // it is time to consume attachment limits.
+        if !summary.attachment_quantities.span.is_empty()
+            && !enforcement.attachments_limits.span.is_active()
+        {
+            enforcement.attachments_limits.span = self
+                .check_attachment_limits(scoping, &summary.attachment_quantities.span)
+                .await?;
+        }
+
+        // Handle trace attachments.
+        if !summary.attachment_quantities.trace.is_empty() {
+            enforcement.attachments_limits.trace = self
+                .check_attachment_limits(scoping, &summary.attachment_quantities.trace)
```

Comment:
As noted above, it should be possible to call `check.apply` once for bytes and once for items.